### DISCUSSION TITLE
Where to start with own extensions?

### DISCUSSION THREAD

**Discussion by marcus-at-localhost** [atrocore/atrocore#178] [Q&A] [answered]
Hi, just wanted to ask where is the best point to start with my own import/export extensions?
Is there some boilerplate code? Or which modules can serve as examples?
Thanks!

**Comment by atrolex** [comment, marked as answer]
Actually, only the modules developed by us appear in the Administration > Module Manager > Store. 

We use Packagist for storing all available modules. In the future we will enable to configure more than one Packagist Repository to pull the modules. So currently you should give us access to your module for us to add it to our Packagist. In the future you would be able to manage your own Packagist Repository.

This is an example how some module can look like:
https://github.com/atrocore/activities-tasks

**Reply by marcus-at-localhost** [reply]
So it's deeply integrated in the composer workflow and I can't just drop in a folder somewhere?
I needed something, where I can download a collection of files in a zip from the DAM (or is that available as plugin?)
Thanks so far.

**Reply by marcus-at-localhost** [reply]
How relevant is the EspoCRM guide to develop Modules, since AtroPIM is based on it? https://docs.espocrm.com/development/modules/

**Reply by rratsun** [reply]
Our module system is completely different from EspoCRM. We don't support any zip extensions because of dependencies. Our main goal is to provide stable updates for the system and the modules, so we have to control versions and dependecies of the system and of the modules. Our system is deeply integrated in the composer workflow.

**Reply by marcus-at-localhost** [reply]
This kind of setup was new to me, thanks for explaining.

Now, is it possible to mass download a bunch of assets at once or is that a case for a custom extension?
